### DISCUSSION TITLE
[wasm] Support wasm plugin to extend UDF and DataSource

### DISCUSSION THREAD

**Discussion by tlightsky** [databendlabs/databend#8506] [Ideas]
**Summary**
Support wasm plugin to extend UDF and DataSource 

something like [this](https://banzaicloud.com/blog/envoy-wasm-filter/)

**Comment by TennyZhuang** [comment]
We finished a UDF demo for TiDB in hackathon 2020 ([A trivial Chinese blog about it](https://blog.zhuangty.com/tidb-hackathon-2020/)),   Wasm is very suitable for the scene. But it’s difficult to design a graceful ABI for complicated types between host language and wasm runtime.

**Comment by CoderYellow** [comment]
maybe your team should refer to apache arrow-datafusion implmentation， is there plan on it
